ngraph/core/reference/src/runtime/reference/pad.cpp

```diff
@@ -16,198 +16,293 @@
 
 #include "ngraph/runtime/reference/pad.hpp"
 
+#include "ngraph/axis_vector.hpp"
+#include "ngraph/check.hpp"
+#include "ngraph/coordinate_index.hpp"
+#include "ngraph/coordinate_transform.hpp"
+
+#include <cassert>
+
 namespace ngraph
 {
     namespace runtime
     {
-        namespace reference
+        namespace impl
         {
-            void pad(const char* data,
-                     const char* pad_value,
-                     char* out,
-                     const size_t elem_size,
-                     const Shape& data_shape,
-                     const Shape& out_shape,
-                     const CoordinateDiff& padding_below,
-                     const CoordinateDiff& padding_above,
-                     const op::PadMode pad_mode)
+            namespace
```

Comment:
General comment: Could you provide a documentation for new structures and functions?

Reply:
I want to have a well documented code.
And if you add 100+ new lines of code without comments it looks bad.
I prefer to have at least the common description why we need to have this code. It shouldn't depend on the code complexity.